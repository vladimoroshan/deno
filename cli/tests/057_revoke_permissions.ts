// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.

const knownPermissions: Deno.PermissionName[] = [
  "run",
  "read",
  "write",
  "net",
  "env",
  "plugin",
  "hrtime",
];

export function assert(cond: unknown): asserts cond {
  if (!cond) {
    throw Error("Assertion failed");
  }
}

function genFunc(grant: Deno.PermissionName): () => Promise<void> {
  const gen: () => Promise<void> = async function Granted(): Promise<void> {
    const status0 = await Deno.permissions.query({ name: grant });
    assert(status0 != null);
    assert(status0.state === "granted");

    const status1 = await Deno.permissions.revoke({ name: grant });
    assert(status1 != null);
    assert(status1.state === "prompt");
  };
  // Properly name these generated functions.
  Object.defineProperty(gen, "name", { value: grant + "Granted" });
  return gen;
}

for (const grant of knownPermissions) {
  Deno.test(genFunc(grant));
}
