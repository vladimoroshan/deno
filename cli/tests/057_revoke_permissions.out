running 7 tests
OK      runGranted [WILDCARD]
OK      readGranted [WILDCARD]
OK      writeGranted [WILDCARD]
OK      netGranted [WILDCARD]
OK      envGranted [WILDCARD]
OK      pluginGranted [WILDCARD]
OK      hrtimeGranted [WILDCARD]

test result: OK 7 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out [WILDCARD]
